directiveAngular/src/app/profilemaking/profilemaking.component.html

```diff
@@ -0,0 +1,65 @@
+<form [formGroup]="ProfileForm" (ngSubmit)="onSubmit()">
+    <label for="fname">First Name: </label>
+    <input id="fname" type="text" formControlName="fname"><br>
+    
+    <div *ngIf="ProfileForm.controls.fname.errors?.['required'] && (ProfileForm.controls.fname.touched||ProfileForm.controls.fname.dirty)">
+        First name is required.
+    </div>
+    <br> <label for="lname">Last Name: </label>
+    <input id="lname" type="text" formControlName="lname"><br>
+    <label for="email">Email: </label>
+    <input id="email" type="text" formControlName="email" #emaill><br>
+   
+    <div *ngIf="ProfileForm.controls.email.errors?.['required'] && (ProfileForm.controls.email.touched||ProfileForm.controls.email.dirty)">
+        Email is required.
+    </div>
+    <div
+        *ngIf="ProfileForm.controls.email.invalid && (ProfileForm.controls.email.touched||ProfileForm.controls.email.dirty) && !ProfileForm.controls.email.errors?.['required']">Email is invalid</div>

```

Comment:
required and invalid message should be different

Reply:
in the first one,if it is touched or dirty and it is not filled,email is required message will be shown.
in the second one,if it is invalid and touched or dirty ,and is filled in ,the invalid message will be shown